shell/Cargo.toml

```diff
@@ -10,6 +10,7 @@ sl-liner = { git = "https://github.com/sl-sh-dev/sl-liner.git" }
 nix = { version = "0.27.1", features = ["signal", "process", "term", "user", "fs", "hostname"] }
 glob = "0.3"
 cfg-if = "1.0"
+builtins = { path = "../builtins" }
```

Comment:
> This is not "correct", the shell crate is intended to encapsulate only the code for a unix shell. It should not depend on builtins or any of the other "lisp" crates. The idea is it can be used stand alone to build a shell (and has an example to do this) with no scripting language at all. As a starting point for a new shell or whatever. Anyway this dep should not be here, the old way of slosh taking the tilde code from here (slosh 100% has to have shell...) seemed fine. What issue where you solving? I know this is not obvious but you should think of shell as one of the foundation crates in the project. If you find having the tilde code here it could go in it's own crate but that seem like overkill and tilde expansion is a shell thing (this is the code that handles parsing stings of "sh").

I see. Ok. so i changed it to make sl_compiler depend on shell. The "real" reason is I needed to move the load builtin to sl_compiler so I could import it into tests in ./slosh/tests/. See the new `new_slosh_vm_with_builtins`

Reply:
Pushing shell down the dep chain is also no good.  Maybe I should not have called it foundation, it is but is also platform dependent.  So by moving it to sl_compiler you have made it impossible to for someone on Windows (or any non-Linux/Mac/maybe generic Unix) to use the language sans the shell part.    I don't think we want to do that, especially not for access to a trivial function.  Currently it should be trivial to build a lang only repl for slosh that is platform independent (that should be feature flag on slosh to exclude the shell for this purpose- not for this PR).  I suggest you either stop the madness and just dup the function or add a new foundation "utils" crate for trivial util functions like this.  I get not wanting two different ways to tilde expand creeping in but this should be very infrequently touched code so am OK with just duping it.

Reply:
@sstanfield understood. I get it.

Went the copying route because that also seems fine given the specific nature of the code.